scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -0,0 +1,579 @@
+
+        if (cmd.hasOption(HELP_OPTION_NAME)) {
+            displayHelp(options);
+            return;
+        }
+
+        action = Action.getAction(cmd);
+
+        if (cmd.hasOption(LOGIN_OPTION_NAME)) {
+            userInfo.setLogin(cmd.getOptionValue(LOGIN_OPTION_NAME));
+        }
+        if (cmd.hasOption(PWD_OPTION_NAME)) {
+            userInfo.setPassword(cmd.getOptionValue(PWD_OPTION_NAME));
+        }
+        if (cmd.hasOption(GROUPS_OPTION_NAME)) {
+            String groupString = cmd.getOptionValue(GROUPS_OPTION_NAME);
+            userInfo.setGroups(groupString.split(","));
+        }
+
+        if (cmd.hasOption(LOGINFILE_OPTION_NAME)) {
+            loginFilePath = cmd.getOptionValue(LOGINFILE_OPTION_NAME);
+        }
+        if (cmd.hasOption(GROUPFILE_OPTION_NAME)) {
+            groupFilePath = cmd.getOptionValue(GROUPFILE_OPTION_NAME);
+        }
+        if (cmd.hasOption(KEYFILE_OPTION_NAME)) {
+            pubKeyPath = cmd.getOptionValue(KEYFILE_OPTION_NAME);
+        }
+
+        if (pubKeyPath == null) {
+            pubKeyPath = getPublicKeyFilePath();
+        }
+        try {
+            pubKey = Credentials.getPublicKey(pubKeyPath);
+        } catch (KeyException e) {
+            exitWithErrorMessage("Could not retrieve public key from '" + pubKeyPath, null, e);
+        }
+
+        switch (action) {
+            case CREATE:
+                nonInteractive = (userInfo.getLogin() != null && userInfo.getPassword() != null && userInfo.getGroups() != null);
+                break;
+            case UPDATE:
+                nonInteractive = (userInfo.getLogin() != null && userInfo.getPassword() != null) || (userInfo.getLogin() != null && userInfo.getGroups() != null);
+                break;
+            case DELETE:
+                nonInteractive = userInfo.getLogin() != null;
+                break;
+        }
+
+        if (!nonInteractive) {
+            askInteractively(console, userInfo, action);
+        }
+
+        updateAccounts(pubKey, userInfo, loginFilePath, groupFilePath, action);
+    }
+
+    /**
+     * Ask the user for parameters missing on the command line
+     */
+    private static void askInteractively(Console console, UserInfo userInfo, Action action) {
+        System.out.println(action);
+        System.out.print("login: ");
+        if (userInfo.getLogin() == null) {
+            userInfo.setLogin(console.readLine());
+        } else {
+            System.out.println(userInfo.getLogin());
+        }
+        System.out.print("password: ");
+        if ((action.isCreate() && userInfo.getPassword() == null) || (action.isUpdate() && userInfo.getPassword() == null && userInfo.getGroups() == null)) {
+            userInfo.setPassword(new String(console.readPassword()));
+        } else {
+            System.out.println("*******");
+        }
+
+        if (action.isCreate() && userInfo.getGroups() == null) {
+            System.out.print("groups for user " + userInfo.getLogin() + ":");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS106)